indigo_app/static/javascript/indigo/views/document_comparisons.js

```diff
@@ -0,0 +1,61 @@
+(function(exports) {
+  "use strict";
+
+  if (!exports.Indigo) exports.Indigo = {};
+  Indigo = exports.Indigo;
+
+  /**
+   * Compare different expressions (in the same language) of a work.
+   */
+  Indigo.DocumentComparisonView = Backbone.View.extend({
+    el: '.document-comparison-view',
+    events: {
+      'click .dismiss': 'dismiss'
+    },
+
+    initialize: function(options) {
+      this.documentContent = options.documentContent;
+
+      $('.menu .comparison').on('click', _.bind(this.show, this));
+
+    },
+
+    comparison: function(comparison_doc_id) {
+
+      var data = {},
+          self = this;
+      data.document = this.documentContent.document.toJSON();
```

Comment:
Lookup the expression being compared with in `Indigo.Preloads.expressions` and set it on a variable on `this`. Then, once it's rendered, you have all the details you need to update the heading. Then the API doesn't need to give you any dates back.

In `initialize`, store the expressions as a backbone collection, which makes it easy to fetch things by id:

```js
this.expressions = new Indigo.Library(Indigo.Preloads.expressions);
```

then here, you can set:

```js
this.compareTo = this.expressions.get(comparison_doc_id)`
```